korge/src/commonMain/kotlin/korlibs/korge/component/StageComponent.kt

```diff
@@ -39,23 +39,25 @@ fun <T : View> T.onAttachDetach(onAttach: Views.(T) -> Unit = {}, onDetach: View
 /**
  * Enables the use of [StageComponent] components.
  */
-fun Views.registerStageComponent() {
+fun Views.registerStageComponent(view: View) {
     val EXTRA_ID = "Views.registerStageComponent"
-    if (views.getExtra(EXTRA_ID) == true) return
-    views.setExtra(EXTRA_ID, true)
+
+    val firstRun = !views.hasExtra(EXTRA_ID)
+    val viewsToTrack = views.extraCache(EXTRA_ID) { linkedSetOf<View>() }
+    viewsToTrack += view
+
+    if (!firstRun) return
+
```

Comment:
I would suggest:

```kotlin
private var Views.viewsToTrack by Extra.Property<LinkedSet<View>> { null }

val firstRun = views.viewsToTrack == null
if (firstRun) {
  views.viewsToTrack = linkedSetOf()
}
...
```

that way it is typed and you don't have to use hasExtra, etc. hasExtra, setExtra, getExtra